contracts/src/components/position.cairo

```diff
@@ -11,6 +11,75 @@ struct Coord {
     y: u32
 }
 
+#[generate_trait]
+impl CoordImpl of CoordTrait {
+
+    // todo@credence revisit this
+    fn interpolate(self: Coord, other: Coord, percent: u32) -> Coord {
```

Comment:
can you explain in a few words the goal of this function ?

Reply:
Something is traveling from A -> B. Given  A, B and the percentage of the journey completed, it calculates the exact coordinate that the travelling entity is at. i.e( somewhere between A and B )

I might rewrite it entirely so no need to look too deeply but that's what it does